discovery/openstack/openstack.go

```diff
@@ -96,12 +98,18 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
+
+	if c.EndpointType != "public" && c.EndpointType != "admin" && c.EndpointType != "internal" {
```

Comment:
You could use the `gophercloud.Availability*` constants directly instead of hard-coded string values.

```go
availability := gophercloud.Availability(c.EndpointType)
if availability != gophercloud.AvailabilityPublic && availability != gophercloud.AvailabilityAdmin && availability != gophercloud.AvailabilityInternal {
    return fmt.Errorf("unknown endpoint_type %q, must be one of %s, %s or %s", c.EndpointType, gophercloud.AvailabilityAdmin, gophercloud.AvailabilityInternal, gophercloud.AvailabilityPublic)
}
```